### PR TITLE
Transit tube update part 1: Construction/access/optimisations

### PR DESCRIPTION
**Pull request by ghost**
[content][sprites]
image

Part 1 of an actual update to these in 8 years, part 2 will be involved with networks and controlling pod speed/maybe traffic.
Also completely removes corner icons (with all code dealing with them gone too) and makes all transit tube sprites 48x48 with a -8 x and y offset to centre them.

## Construction
* Metal recipes > Transit tube parts.
* Wrench in place. (Except pods)
* (Pods/stations only) Add transit pod circuitboard/station airlock electronics.
* Add reinforced glass to finish. 

## Deconstruction
* Welding tool on part.
* Unwrench from place. (Except pods)
* (Pods/stations only) Crowbar to pry out circuitboard.
* Weld to dismantle back into sheets.

:cl:
 * rscadd: Transit tubes are now buildable from metal stack recipes and deconstructable, along with their stations and pods. Requires anchoring and plating with reinforced glass to finish these, plus a circuitboard from R&D for the transit pods and airlock electronics for the stations. Welding transit tube parts turns them into frames again.
 * rscadd: Transit tube stations now have access requirements support, and airlock electronics to edit this. All transit tube stations on maps have been updated with suitable access requirements to match their rooms.
 * tweak: Transit tube corners are no longer separate objects.
 * tweak: Transit tubes move faster now.
 * soundadd: Transit tube station doors now have opening and closing sounds.

### CODE REVIEW

**Comment by GreatBigFailure** [comment]
Oh my yes

**Comment by SonixApache** [comment]
i hate tubes

adding these for player construction will have unforeseen consequences

File: code/game/objects/structures/transit_tubes.dm [hunk 1]
```diff
@@ -672,4 +703,18 @@
 		else
 	return
 
+/obj/structure/transit_tube/proc/iconstate2framedir(var/icon_state)
```
Comment (DamianX) [inline code/game/objects/structures/transit_tubes.dm]: looks like the `icon_state` parameter is always equal to `src.icon_state`. Change all callers as well.
```suggestion
/obj/structure/transit_tube/proc/iconstate2framedir()
```

File: code/game/objects/structures/transit_tubes.dm [hunk 2]
```diff
@@ -196,6 +263,7 @@
 
 /obj/structure/transit_tube/station/proc/close_animation()
 	if(icon_state == "open")
+		playsound(get_turf(src), 'sound/machines/windowdoor.ogg', 50, 1)
```
Comment (DamianX) [inline code/game/objects/structures/transit_tubes.dm]:
```suggestion
		playsound(src, 'sound/machines/windowdoor.ogg', 50, 1)
```

File: code/game/objects/structures/transit_tubes.dm [hunk 3]
```diff
@@ -186,6 +252,7 @@
 
 /obj/structure/transit_tube/station/proc/open_animation()
 	if(icon_state == "closed")
+		playsound(get_turf(src), 'sound/machines/windowdoor.ogg', 50, 1)
```
Comment (DamianX) [inline code/game/objects/structures/transit_tubes.dm]:
```suggestion
		playsound(src, 'sound/machines/windowdoor.ogg', 50, 1)
```

File: code/game/objects/structures/transit_tube_assembly.dm [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        playsound(get_turf(src), 'sound/items/Deconstruct.ogg', 50, 1)
+        to_chat(user, "<span class='notice'>You begin to add reinforced glass to \the [src]...</span>")
+        if(G.amount < 2)
+            to_chat(user, "<span class='warning'>You need 2 sheets of glass to do this.</span>")
+            return 1
+        if(do_after(user,src,40))
+            if(G.amount < 2) //User being tricky
+                return 1
+            G.use(2)
+            to_chat(user, "You add the reinforced glass to the [src].")
```
Comment (DamianX) [inline code/game/objects/structures/transit_tube_assembly.dm]:
```suggestion
            to_chat(user, "<span class='notice'>You add the reinforced glass to \the [src].</span>")
```

File: code/game/objects/structures/transit_tubes.dm [hunk 5]
```diff
@@ -119,12 +131,66 @@
...
+					TTS.electronics.dir_access = req_access_dir
+					TTS.electronics.access_nodir = access_not_dir
+			qdel(src)
+		return 1
+
+/obj/structure/transit_tube_pod/attackby(obj/item/W as obj, mob/user as mob)
+	if(iswelder(W))
+		var/obj/item/tool/weldingtool/WT = W
+		to_chat(user, "You begin to cut the glass off...")
+		if(WT.do_weld(user, src, 40))
```
Comment (DamianX) [inline code/game/objects/structures/transit_tubes.dm]:
```suggestion
		if(WT.do_weld(user, src, 4 SECONDS))
```

File: code/game/objects/structures/transit_tubes.dm [hunk 6]
```diff
@@ -119,12 +131,66 @@
...
+						TTS.electronics.one_access = 1
+					TTS.electronics.dir_access = req_access_dir
+					TTS.electronics.access_nodir = access_not_dir
+			qdel(src)
+		return 1
+
+/obj/structure/transit_tube_pod/attackby(obj/item/W as obj, mob/user as mob)
+	if(iswelder(W))
+		var/obj/item/tool/weldingtool/WT = W
+		to_chat(user, "You begin to cut the glass off...")
```
Comment (DamianX) [inline code/game/objects/structures/transit_tubes.dm]:
```suggestion
		to_chat(user, "<span class='notice'>You begin to cut the glass off...</span>")
```

**Comment by adacovsk** [comment]
I tried implementing them in my map, but they were such a pain because of those little corner pieces. This will be a good update, if just for that alone. Hopefully they go as fast as disposals or faster, also, making them under floors

File: code/game/objects/structures/transit_tubes.dm (hunk 1 above)
Comment (ghost) [reply code/game/objects/structures/transit_tubes.dm]: Fixed

File: code/game/objects/structures/transit_tubes.dm (hunk 3 above)
Comment (ghost) [reply code/game/objects/structures/transit_tubes.dm]: Fixed

File: code/game/objects/structures/transit_tubes.dm (hunk 5 above)
Comment (ghost) [reply code/game/objects/structures/transit_tubes.dm]: Fixed

File: code/game/objects/structures/transit_tubes.dm (hunk 2 above)
Comment (ghost) [reply code/game/objects/structures/transit_tubes.dm]: Fixed

File: code/game/objects/structures/transit_tube_assembly.dm (hunk 4 above)
Comment (ghost) [reply code/game/objects/structures/transit_tube_assembly.dm]: Fixed

File: code/game/objects/structures/transit_tubes.dm (hunk 6 above)
Comment (ghost) [reply code/game/objects/structures/transit_tubes.dm]: Fixed

File: code/game/objects/structures/transit_tubes.dm [hunk 7]
```diff
@@ -672,4 +703,18 @@
 		else
 	return
 
+/obj/structure/transit_tube/proc/iconstate2framedir()
+	switch(icon_state)
+		if("N-S","NE-SW","N-SW","N-SE","N-SW-SE","N-SE-SW","N-S-pass")
+			return 1
```
Comment (ShiftyRail) [inline code/game/objects/structures/transit_tubes.dm]: NORTH

File: code/game/objects/structures/transit_tubes.dm [hunk 8]
```diff
@@ -672,4 +703,18 @@
...
+		if("S-NE","S-NW","S-NE-NW","S-NW-NE")
+			return 2
```
Comment (ShiftyRail) [inline code/game/objects/structures/transit_tubes.dm]: SOUTH

File: code/game/objects/structures/transit_tubes.dm [hunk 9]
```diff
@@ -672,4 +703,18 @@
...
+		if("E-W","NW-SE","E-NW","E-SW","E-NW-SW","E-SW-NW","E-W-pass")
+			return 4
```
Comment (ShiftyRail) [inline code/game/objects/structures/transit_tubes.dm]: EAST

File: code/game/objects/structures/transit_tubes.dm [hunk 10]
```diff
@@ -672,4 +703,18 @@
...
+		if("W-SE","W-NE","W-SE-NE","W-NE-SE")
+			return 8
```
Comment (ShiftyRail) [inline code/game/objects/structures/transit_tubes.dm]: WEST

File: code/game/objects/structures/transit_tubes.dm (hunk 7 above)
Comment (ghost) [reply code/game/objects/structures/transit_tubes.dm]: Fixed

File: code/game/objects/structures/transit_tubes.dm (hunk 8 above)
Comment (ghost) [reply code/game/objects/structures/transit_tubes.dm]: Fixed

File: code/game/objects/structures/transit_tubes.dm (hunk 9 above)
Comment (ghost) [reply code/game/objects/structures/transit_tubes.dm]: Fixed

File: code/game/objects/structures/transit_tubes.dm (hunk 10 above)
Comment (ghost) [reply code/game/objects/structures/transit_tubes.dm]: Fixed

**Comment by adacovsk** [comment]
What about adding them to the RPD?

**Comment by Eneocho** [comment]
What if you could connect a high pressure canister/atmos device to the transit tubes to make them move faster? It'd be neat and would give a use to the ultratanks that atmosians sometimes make.

**Pull request merged**